pkg/reconciler/revision/background.go

```diff
@@ -160,7 +164,16 @@ func (r *backgroundResolver) Resolve(rev *v1.Revision, opt k8schain.Options, reg
 	}
 
 	ret := r.results[name]
-	return ret.statuses, ret.err
+	if ret.err != nil {
+		return nil, ret.err
+	}
+
+	array := make([]v1.ContainerStatus, len(rev.Spec.Containers))
+	for k, v := range ret.statuses {
+		array[k] = v
+	}
```

Comment:
I guess we don't **really** care about the extra cost of creating the objects here (especially since they're mostly going to be small anyway), but just to record it: Would we achieve the same if we guarded the `result.remaining` below in a `if result.statuses[item.index].resolveDigest != ""` check?

Reply:
yeah I think that'd work too, it just seems a little nicer/less error-prone in general/easier to debug to avoid the bookkeeping of maintaining `remaining` and focus on whether the _actual_ state is what we want to determine if we're done. As you say, there isn't really a big memory/efficiency concern anyway, things will be pretty dominated by all the network activity, and none of this is in request path. (That said I don't feel strongly and would totally do it the other way if you did particularly care).